samples/excel/99-just-for-fun/gradient.yaml

```diff
@@ -41,29 +40,68 @@ script:
                 }
             });
 
-            let originalSize = parseInt($('#size').val() as string);
+            let originalSize = parseInt($("#size").val() as string);
 
             // Create a 2D in-memory array to hold the colors
             const colors2D = Array(originalSize);
             for (let row = 0; row < originalSize; row++) {
                 colors2D[row] = Array(originalSize);
             }
 
-            fillInColors(colors2D, originalSize);
-
-            let range = sheet.getCell(1, 1).untrack().getResizedRange(
-                originalSize - 1, originalSize - 1).untrack();
-            for (let row = 0; row < originalSize; row++) {
-                for (let col = 0; col < originalSize; col++) {
-                    range.getCell(row, col).untrack().format.fill.color =
-                        colors2D[row][col];
+            if (Office.context.requirements.isSetSupported("ExcelApi", 1.9)) {
+                // ExcelApi 1.9 introduced the setCellProperties APIs to quickly set different properties across a range.
+                fillInColorsWithCellProperties(colors2D, originalSize);
+                sheet.getRangeByIndexes(1, 1, originalSize, originalSize).setCellProperties(colors2D);
+            } else { 
+                fillInColors(colors2D, originalSize);
+
+                let range = sheet.getCell(1, 1).untrack().getResizedRange(originalSize - 1, originalSize - 1).untrack();
+                for (let row = 0; row < originalSize; row++) {
+                    for (let col = 0; col < originalSize; col++) {
+                        range.getCell(row, col).untrack().format.fill.color = colors2D[row][col];
+                    }
                 }
             }
 
             sheet.activate();
             await context.sync();
         }
 
+        function fillInColorsWithCellProperties(colors2D: Excel.SettableCellProperties[][], size: number) {
+            const topColors = getColorsArray(
```

Comment:
I think the code would be much cleaner if you just re-use the `fillInColors` function and then `.map` over it to create the data structure you want.  Otherwise, you are copying the code even though 90% of it could have been re-usable...

Also, I think the creation of the `colors2D` array of the original could code really move into this function (making it a `create2DColorsArray` function).  That would further clean up and consolidate the logic.